Omit extra libraries because their interfaces aren't exposed to applications
by libkrb5, unless do_deps is set to 1, which indicates that the caller
wants the whole list.

diff -up krb5-1.7/src/krb5-config.in krb5-1.7/src/krb5-config.in
--- krb5-1.7/src/krb5-config.in	2009-06-04 14:34:33.000000000 -0400
+++ krb5-1.7/src/krb5-config.in	2009-06-04 14:34:33.000000000 -0400
@@ -220,7 +220,11 @@ if test -n "$do_libs"; then
     fi
 
     if test $library = 'krb5'; then
-	lib_flags="$lib_flags -lkrb5 -lk5crypto -lcom_err $GEN_LIB $LIBS $SELINUX_LIBS $DL_LIB"
+	if test 0$do_deps -eq 1 ; then
+	    lib_flags="$lib_flags -lkrb5 -lk5crypto -lcom_err $GEN_LIB $LIBS $SELINUX_LIBS $DL_LIB"
+	else
+	    lib_flags="$lib_flags -lkrb5 -lk5crypto -lcom_err"
+	fi
     fi
 
     echo $lib_flags
